src/commands.rs

```diff
@@ -59,20 +61,49 @@ impl Commands {
     }
 
     pub(crate) fn execute<'m>(&'m self, cx: Context, msg: Message) {
-        if !msg.is_own(&cx) && msg.content.starts_with(PREFIX) {
-            let message = &msg.content.clone();
-            self.state_machine.process(&message).map(|matched| {
-                info!("Executing command {}", message);
-                let args = Args {
-                    cx,
-                    msg,
-                    params: matched.params,
-                };
-                if let Err(e) = (matched.handler)(args) {
-                    println!("{}", e);
+        if !msg.is_own(&cx) {
+            for message in find_commands(&msg.content) {
```

Comment:
So this searches every message of every channel for the substring (.  Ideally I would prefer the bot to not parse messages that are not meant for it, hence the `?` prefix for commands.

Reply:
the bot already receives every message in every channel - i don't think iterating though each message one more time requires a lot of resources. 

ive seen a lot of bots where inline commands use @-mentions instead of a sigil if we want to go that way, we won't have to parse every message

Reply:
basically as it works right now the bot reads the first character of every message checking if it starts with `?`.  These changes would mean every message would be read more or less in its entirety.

Reply:
That is the trade off. Because the inline command could be anywhere in the message, the entire message has to be parsed, which means scanning at most 2000 characters. The Discord isn't super active as a whole, so it might be doing 3-4 per second, at most, which means the chances of 2 messages being scanned concurrently is slim to none.